dms/models/directory.py

```diff
@@ -474,7 +464,15 @@ def _compute_directory_type(self):
             if record.is_root_directory:
                 record.parent_id = None
             else:
-                record.root_storage_id = None
+                record.storage_id = None
+                if self.env.context.get(
+                    "active_model"
+                ) == self._name and self.env.context.get("active_id"):
+                    record.parent_id = self.env.context.get("active_id")
```

Comment:
Why are you assigning `parent_id` in a method called `_compute_directory_type`, and mixing also `storage_id`?

Reply:
Ok, i have changed compute function name and remove related to `storage_id`.